### ISSUE TITLE
Support Result.isOk and Result.isError

### ISSUE THREAD

**Issue by zprobinson** [fable-compiler/Fable#3726]
### Description

JavaScript/TypeScript Fable does not support neither Result.isOk or Result.isError.

### Repro code

```fs
open Feliz

let validateEmpty (s: string) =
  if s.Trim().Length = 0 then
    Error "This field is required"
  else
    Ok ""

[<ReactComponent>]
let Form () =
  let username, setUsername = React.useState ""

  let validInput = username |> validateEmpty |> Result.isOk

  printfn "it'll be null: %b" validInput

  Html.form [
    prop.onSubmit (fun _ -> printfn "you did it")
    prop.children [
      Html.input [ prop.type' "text"; prop.value username; prop.onChange setUsername ]
      Html.button [ prop.type' "submit"; prop.disabled (not validInput); prop.children [ Html.text "Click" ] ] ]
```

Please provide the F# code to reproduce the problem or a link to the [REPL](https://fable.io/repl/).
Ideally, it should be possible to easily turn this code into a unit test.

### Expected and actual results

Please provide the expected and actual results.
Expected: a pretty disabled button

Actual:
This error message in the console 
`client: ./Pages/Login/View.fs(63,54): (63,68) error FABLE: Microsoft.FSharp.Core.ResultModule.IsError (static) is not supported by Fable`

### Related information

* Fable version: `dotnet fable --version`
   * 4.1.4 from CLI
   * 4.2 based on Paket for Fable.Core
* Operating system
   * MacOS 14.2.1

**Comment by gopalchobey9** [comment]
@zprobinson  can  i work on this ?

**Comment by zprobinson** [comment]
@gopalchobey9 I don't mind. I'm taking a look at it as well, but I'm not very familiar with this repo as of yet.

**Comment by zprobinson** [comment]
@gopalchobey9 I've made an MVP PR at # #3727 I'm open to collaboration for updates.

**Comment by zprobinson** [comment]
Resolved by #3727 so I will close this issue.
